src/app/services/cog.service.ts

```diff
@@ -0,0 +1,393 @@
+/*
+ * Licensed to Gisaïa under one or more contributor
+ * license agreements. See the NOTICE.txt file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Gisaïa licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { getTitilerPreviewUrl, VisualisationPreview } from 'app/tools/cog';
+import { getItem } from 'app/tools/utils';
+import { Expression, Filter, Search } from 'arlas-api';
+import {
+  CogModalComponent, CogVisualisationData, DataGroup, ItemDataType, ResultlistNotifierService, VisualisationInterface
+} from 'arlas-web-components';
+import { Action, ActionFilter, ElementIdentifier, ResultListContributor } from 'arlas-web-contributors';
+import { projType } from 'arlas-web-core';
+import { ArlasCollaborativesearchService } from 'arlas-wui-toolkit';
+import { first, map, Observable, Subject, take } from 'rxjs';
+import { ActionManagerService } from './action-manager.service';
+import { ContributorService } from './contributors.service';
+import { VisualizeService } from './visualize.service';
+
+/**
+ * This service is the interface between the ResultlistService and the VisualizeService for the viusalisation of COGs.
+ * It stores the state of the visualisation as well as offers utilitary methods to facilitate those visualisations.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class CogService<L, S, M> {
+  /** --- Configuration of the current resultlist */
+  /** Input field of the contributor configuration */
+  public contributorConfig: any;
+  /** Contributor Id */
+  public contributorId: string;
+  /** Current configuration for the visualisation of COGs */
+  public currentCogVisualisationConfig: VisualisationInterface[];
+
+  /** --- Visualisation state */
+  /** Current COG visualisation */
+  private readonly selectedCogVisualisation = new Map<string, VisualisationPreview>();
+  /** Emits any change to the visualisation used for the COGs */
+  public cogVisualisationChange = new Subject<VisualisationPreview>();
```

Comment:
could you use the tuple  
cogVisualisationChangeSource as a subject
&
cogVisualisationChange$ as an observable
?
It makes it clear for subject use in services + we can complete more safely the observable